docs/adr/0019-use-1password-for-sharing-secrets.md

```diff
@@ -0,0 +1,96 @@
+  secrets with other Truss personnel who are not assigned to the project.
+* 1Password also has familiar and user-friendly graphical interfaces via the
+  web and a multi-platform desktop application.
+* Access to secrets can be revoked as quickly and as easily as it was granted.
+
+Consequences:
+
+* We will need to create a vault, which takes time and costs money.
+* Some synchronization will be required on how to use the vault and what
+  secrets to store in it.
+* Secrets that are required by the easi-backend app will have to be updated in
+  SSM anyway, which means changing them requires updating the value in two
+  places (1Password and SSM).
+
+## Pros and Cons of the Alternatives
+
+### Email
+
+* `+` Fast.
+* `+` Some of our peers share secrets with us using email, so forwarding the
+  emails is an easy and straightforward way to distribute secrets.
+* `-` Poor access controls: once an email is sent, its contents cannot be recalled.
+* `-` Emails can be easily shared with a wide audience inadvertently, resulting
+  in a pervasive risk of uncontrolled disclosure.
+* `-` If we have to manage any secrets that cannot be rotated, there will be no
+  way to guarantee that sent emails are burned after reading: rather, they will
+  likely live on in the archived email of whoever received them.
+* `-` There would be no central source of truth on what the current secrets
+  are, which complicates onboarding/off-boarding, engineers returning from
+  vacation, etc.
+
+### Slack DMs
+
+* `+` It's easy to find people in Slack, and we all use it.
+* `+` Only two people are in a DM thread. Unlike email, it is not easy to share
+  contents of a DM inadvertently.
+* `-` Due to CMS record-keeping policies, DMs cannot be deleted and may be
+  subject to subpoena or FOIA requests, which makes them public records.
+* `-` There is no central source of truth.
+
+### 1Password
+
+* `+` Per-user access controls.
+* `+` We can restrict the vault to project engineers according to the principle
+  of least access.
+* `+` The 1Password vault is a central source of truth on secrets.
+* `-` We have to create a vault, which costs money. May require CORS approval.
+* `-` Some secrets will have to be updated in 1Password and SSM when they
+  change, such as API keys.
+
+### AWS SSM / ParameterStore
+
+_See also:_ [Use AWS SSM Parameter Store for Config/Secrets management](./0008-ssm-for-configsecrets.md)
+
+* `+` Provides a _single_ source of truth: Updates in SSM are automatically
+  shared with the easi-backend app.
+* `+` We already depend on ParameterStore to store secrets for the app, so
+  there is no new point of potential security failure.
+* `-` Spin-up for a new engineer on gaining access to SSM is time consuming and
+  complex.
+* `-` Neither the web console nor the cli interface are user-friendly and
+  intuitive to the uninitiated, especially since both require authentication
+  via CloudTamer.
```

Comment:
Could this be improved via training/documentation?
```suggestion
* `-` Neither the web console nor the cli interface are user-friendly and
  intuitive to the uninitiated.
```

Reply:
Not likely. App engineers don't have to get on the VPN as much as we do, so the VPN is the big lift there, and that's always going to be annoying.

Reply:
Will app engineers need to update SSM? Is there currently a way to update SSM without getting on the VPN? Even if we use 1Password, the answers to those two questions may suggest there's still a remaining problem to be solved.

Reply:
> Will app engineers need to update SSM?

User story: "As an app engineer, I want to update the value of a secret in SSM and also distribute it to the other app engineers."

In this situation, one engineer will need to update SSM. (That's already the case and it's not changing.) Depending on which secret it is, they will also have to update 1Password, and then all engineers will then have to update their local development environment by retrieving the secret from 1Password. (That's new.)

Using 1Password reduces the number of times app engineers must collectively connect to SSM, and hence the VPN, from once each to once in total.

> Is there currently a way to update SSM without getting on the VPN?

I am not aware of a way to access SSM without ever connecting to the VPN, no.

Reply:
92a30fc